app/src/main/java/com/omen273/crossLingo/GameActivity.kt

```diff
@@ -576,7 +573,8 @@ class GameActivity : AppCompatActivity(), CrosswordView.OnLongPressListener,
         }
 
     override fun onCrosswordSolved(view: CrosswordView) {
-        star_number.text = (star_number.text.toString().toInt() + BONUS_ON_SOLVE).toString()
+        star_number.text = ((star_number.text.toString().toIntOrNull() ?: starNumber) + BONUS_ON_SOLVE).toString()
```

Comment:
Please check that you have right number of stars if you solve crossword with help of non-free tips if star indicator is free. I think you need to update starNumber in this case

Reply:
And add test on it

Reply:
It should work so please add or modify test in separate ticket